### ISSUE TITLE
Lack of yamllint check

### ISSUE THREAD

**Issue by brandor5** [snapp/template-ansible-collection#2]
While testing the template I noticed there is no run for yamllint. Is this something you'd like to include in the pre-commit hooks?

~~~
 yamllint .
./galaxy.yml
  24:81     error    line too long (82 > 80 characters)  (line-length)
  29:81     error    line too long (108 > 80 characters)  (line-length)

./.github/settings.yml
  3:81      error    line too long (156 > 80 characters)  (line-length)
  18:81     error    line too long (94 > 80 characters)  (line-length)
  21:81     error    line too long (133 > 80 characters)  (line-length)
  24:81     error    line too long (94 > 80 characters)  (line-length)
  27:81     error    line too long (90 > 80 characters)  (line-length)
  30:81     error    line too long (119 > 80 characters)  (line-length)
  83:81     error    line too long (114 > 80 characters)  (line-length)
  91:81     error    line too long (269 > 80 characters)  (line-length)
  93:81     error    line too long (86 > 80 characters)  (line-length)
  97:81     error    line too long (91 > 80 characters)  (line-length)
  99:81     error    line too long (174 > 80 characters)  (line-length)
  103:81    error    line too long (98 > 80 characters)  (line-length)
  105:81    error    line too long (94 > 80 characters)  (line-length)
  107:81    error    line too long (162 > 80 characters)  (line-length)

./.github/workflows/ansible-lint.yml
  6:1       warning  truthy value should be one of [false, true]  (truthy)
  12:81     error    line too long (135 > 80 characters)  (line-length)

./.github/workflows/ansible-test.yml
  3:81      error    line too long (110 > 80 characters)  (line-length)
  21:1      warning  truthy value should be one of [false, true]  (truthy)
  22:81     error    line too long (88 > 80 characters)  (line-length)
  28:81     error    line too long (160 > 80 characters)  (line-length)
  63:9      warning  comment not indented like content  (comments-indentation)
  94:81     error    line too long (87 > 80 characters)  (line-length)
  102:9     warning  comment not indented like content  (comments-indentation)
  128:81    error    line too long (94 > 80 characters)  (line-length)
  147:81    error    line too long (93 > 80 characters)  (line-length)
  188:11    warning  comment not indented like content  (comments-indentation)

./.github/workflows/antsibull-docs-lint.yml
  2:81      error    line too long (116 > 80 characters)  (line-length)
  6:1       warning  truthy value should be one of [false, true]  (truthy)
  12:81     error    line too long (165 > 80 characters)  (line-length)

./changelogs/changelog.yml
  1:1       warning  missing document start "---"  (document-start)

./changelogs/config.yml
  1:1       warning  missing document start "---"  (document-start)
  12:1      error    wrong indentation: expected at least 1  (indentation)

./docs/links.yml
  2:81      error    line too long (98 > 80 characters)  (line-length)
  4:81      error    line too long (84 > 80 characters)  (line-length)
  5:81      error    line too long (84 > 80 characters)  (line-length)
  6:81      error    line too long (95 > 80 characters)  (line-length)
  11:81     error    line too long (93 > 80 characters)  (line-length)
  12:81     error    line too long (93 > 80 characters)  (line-length)
  13:81     error    line too long (87 > 80 characters)  (line-length)
  14:81     error    line too long (97 > 80 characters)  (line-length)
  17:81     error    line too long (83 > 80 characters)  (line-length)
  18:81     error    line too long (85 > 80 characters)  (line-length)
  21:81     error    line too long (83 > 80 characters)  (line-length)
  32:81     error    line too long (84 > 80 characters)  (line-length)
  35:3      warning  comment not indented like content  (comments-indentation)
  38:81     error    line too long (83 > 80 characters)  (line-length)
  39:81     error    line too long (90 > 80 characters)  (line-length)
~~~

**Comment by snapp** [comment]
Thanks @brandor5!

For sure, I'm a huge fan of letting the tools work for us to maintain code cleanliness and I simply missed yamllint when I was hammering the initial implementation out.

I think the path forward would be to add something like the following to the `.pre-commit-config.yaml` file

```
 - repo: https://github.com/adrienverge/yamllint.git
   sha: v1.32.0
   hooks:
     - id: yamllint
       args: ['-d {extends: relaxed, rules: {line-length: disable}}', '-s']
```

I tend to prefer more opinionated/strict settings, so please take a look at the `args` part of the example I tossed in to find something that would be more appropriate.

Only other thing I can think of is that you should probably add it to `{{cookiecutter.namespace}}/{{cookiecutter.collection_name}}/.pre-commit-config.yaml` as well, along with any necessary file exclusions to avoid getting false positives from some of the project contents.

--
Snapp

**Comment by snapp** [comment]
Resolved by #3
